babel/src/main/java/org/apache/calcite/sql/babel/SqlExecMacroParam.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.babel;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlSpecialOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+
+/**
+ * Parse tree for {@code SqlExecMacroParam} SqlCall.
+ */
+public class SqlExecMacroParam extends SqlCall {
+  private static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("PARAM EQUAL", SqlKind.OTHER);
+
+  private final SqlNode name;
+  private final SqlNode value;
+
+  /**
+   * Create an {@code SqlExecMacroParam} without name.
+   *
+   * @param pos  Parser position, must not be null
+   * @param value  Value of the parameter
+   */
+  public SqlExecMacroParam(SqlParserPos pos, SqlNode value) {
```

Comment:
Can we get rid of this constructor and just use the one below with "name" optionally null?

Reply:
Code wise, getting rid of this constructor would look like removing duplicate. However, the point of having two constructor taking different number of parameters is to enforce the name to be null internally. We don't need to rely on the future SWE to follow the standard of passing a null for name when initializing one without name. Please let me know what you think. If the pro is greater than the con, we should still get  rid of the constructor.

Reply:
I think either way is fine, so I'll leave it up to you to decide whichever you'd prefer.

However, if you do choose to keep two different constructors, you should only have one main constructor which the others call. So in this case, that would mean having the SqlExecMacroParam(pos, value) constructor call the SqlExecMacroParam(pos, name, value) one.

Reply:
Good suggestion. Will do.